core/src/main/java/nav/statusplattform/core/openingHours/ServicesUpTimeRenumerator.java

```diff
@@ -0,0 +1,248 @@
+        */
+
+        long totalMinsBetweenLastRecordAndTo = 0;
+
+        ZonedDateTime currentTime = ZonedDateTime.now();
+        int noOfDays = (int) noOfDaysBetweenTwoPeriods(lastRecord.getCreated_at(), to);
+
+
+        //The total of Opening hours start and end time in minutes
+        if (noOfDays > 1) {
+            totalMinsBetweenLastRecordAndTo = ChronoUnit.MINUTES.between(lastRecord.getCreated_at().toLocalTime(), to.toLocalTime()) * noOfDays;
+        }
+
+
+        /*Calculates minutes from the last record start time if its time is greater than the opening hours start time */
+        long minsToLastRecordActualStart = 0;
+
+        if (withinWorkingHours(lastRecord, from, to)) {
+            if (currentTime.toLocalTime().isBefore(to.toLocalTime())) {
+                minsToLastRecordActualStart = ChronoUnit.MINUTES.between(lastRecord.getCreated_at().toLocalTime(), currentTime.toLocalTime());
+            } else {
+                minsToLastRecordActualStart = ChronoUnit.MINUTES.between(lastRecord.getCreated_at().toLocalTime(), to.toLocalTime());
+            }
+        }
+
+
+        //Last record total minutes
+        long expectedUptimeLastRecord = totalMinsBetweenLastRecordAndTo - minsToLastRecordActualStart;
+        //long expectedUptimeLastRecord = minsFromStartToLastRecordActualStart;
+
+        //if the last record is up, its time is included in the calculation
+        long actualUpTimeLastRecord = 0L;
+        if (lastRecord.getStatus() == ServiceStatus.OK) {
+            actualUpTimeLastRecord = expectedUptimeLastRecord;
+        }
+
+        //total expected time
+        long expectedUptimeTotal = sumOfExpectedUptime + expectedUptimeLastRecord;
+
+        //total actual up time
+        long actualUpTimeTotal = sumOfActualUptime + actualUpTimeLastRecord;
+
+        uptimeTotal = new UpTimeTotal(actualUpTimeTotal, expectedUptimeTotal);
+    }
+
+    private long noOfDaysBetweenTwoPeriods(ZonedDateTime periodOne, ZonedDateTime periodTwo) {
+        // Returns the sum of Opening Hours times between the records previous and current times.
+        return (int) (ChronoUnit.DAYS.between(periodOne, periodTwo));
+    }
+
+    /*Returns true if the time record is created is greater than the opening hours start time and is less than the end times*/
+    private boolean withinWorkingHours(RecordEntity recordEntity, ZonedDateTime from, ZonedDateTime to) {
+        // Has a value of  0 or less if opening hours start time is before or equal to the record start time
+        int withinStartTime = from.toLocalTime().compareTo(recordEntity.getCreated_at().toLocalTime());
+
+        // Has a value greater than zero if opening hours end time is equal or after the record end time
+        int withinEndTime = to.toLocalTime().compareTo(recordEntity.getCreated_at().toLocalTime());
+
+        //Returns true if record is within opening hours, otherwise false.
+        return withinStartTime <= 0 && withinEndTime >= 0;
+    }
```

Comment:
The entity referred to as `OpeningHours` in the code is never involved in this method. What am I missing?